File: sdk/typescript/src/blockStep.ts
import { BlockedState, DependsOn, Field } from "./types";

export interface BlockStep {
    block: string;

    allow_dependency_failure?: boolean;
    blocked_state?: BlockedState;
    branches?: string[] | string;
    depends_on?: Array<DependsOn | string> | null | string;
    fields?: Field[];
    id?: string;
    identifier?: string;
    if?: string;
    key?: string;
    label?: string;
    name?: string;
    prompt?: string;
}

File: sdk/csharp/src/Buildkite.Sdk/BlockStep.cs
using System.Collections.Generic;

namespace Buildkite.Sdk
{
    public class BlockStep : PipelineStep
    {
        public string? Block { get; set; }
        public string? Prompt { get; set; }
        public List<Field>? Fields { get; set; }
        public BlockedState? BlockedState { get; set; }

        public BlockStep()
        {
        }

        public BlockStep(string block)
        {
            Block = block;
        }

        public BlockStep(string label, string block)
        {
            Label = label;
            Block = block;
        }
    }
}
